### PR TITLE
Implement Telegram bot support

### PR DESCRIPTION
**Pull request by forestlovewood**
This PR introduces support for Telegram bot notifications.

### CODE REVIEW

File: internal/alert/telegram.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+type TelegramPayload struct {
+	ChatID string `json:"chat_id"`
+	Text   string `json:"text"`
+}
+
+//SendTelegramMessage Sends a notification message to a Telegram Webhook.
+func SendTelegramMessage(item string, config config.TelegramConfig, client *http.Client) error {
+	content := fmt.Sprintf("%s Ready for Purchase", item)
+
```
Comment (ianmarmour) [inline internal/alert/telegram.go:22]: Nit remove extra whitespace line.

File: internal/alert/telegram.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	req, _ := http.NewRequest("POST", fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", config.APIKey), bytes.NewBuffer(payload))
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, _ := client.Do(req)
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 400 {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatal(err)
```
Comment (ianmarmour) [inline internal/alert/telegram.go]: Probably don't want to log.Fatal here, I would prefer to log.Println and then pass the error back up. I know it's not incredibly idiomatic but if a user has sms/discord/telegram if just one fails due to some outside error I would like the program to continue.

File: internal/alert/telegram.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		}
+		bodyString := string(bodyBytes)
+
+		fmt.Println(bodyString)
+		fmt.Println("Unable to send message to telegram, bad request.")
```
Comment (ianmarmour) [inline internal/alert/telegram.go]: Prefer log.Println (it has timestamping)

File: internal/alert/telegram.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		return errors.New("Bad Request")
```
Comment (ianmarmour) [inline internal/alert/telegram.go:36]: Maybe make the message a little more indicative of the failure state.

File: internal/browser/browser.go [hunk 5]
```diff
@@ -111,6 +112,13 @@ func GetInventoryStatus(ctx context.Context, sku string, locale string) (*Invent
 
 	inventoryStatus := InventoryStatus{}
 
+	if strings.HasPrefix(string(stockResponseBody), "Rate limit exceeded") {
```
Comment (ianmarmour) [inline internal/browser/browser.go]: We don't need to use strings there we can check for the response status code 429 instead.

**Review by ianmarmour — CHANGES_REQUESTED** [review]
Mostly minor stuff here, you'll need to go ahead and check the status code for 429 but I believe I check for 500's somewhere so you can use that as an example.

File: internal/browser/browser.go (hunk 5 above)
Comment (ianmarmour) [reply internal/browser/browser.go]: Side note I don't know how happy I am with this logic overall, I'm assuming during the times when stock is active we will end up getting these frequently but if we backup for a second there might be a chance that the user doesn't get a card at all.... thoughts?

Comment (forestlovewood) [reply internal/browser/browser.go]: @ianmarmour Good point about that... stripped out the logic again.

File: internal/alert/telegram.go (hunk 3 above)
Comment (forestlovewood) [reply internal/alert/telegram.go]: Agreed, I blame that on rushing because I'm at work. That entire section of code was a test and has been removed.

**Pull request merged**